Add release notes for task performance issue (#28390)
gradle/gradle: platforms/documentation/docs/src/docs/release/notes.md

Additionally, for JUnit5 and Spock, we updated the test descriptor for dynamic and parameterized tests to include information about the class name and method name containing the test.
These enhancements enable IDEs to offer improved navigation and reporting capabilities for dynamic and parameterized tests.

#### Fix IDE performance issues with large projects

> A performance issue in the tooling API that caused delays at the end of a task execution with large projects was found and fixed. 

Reviewer: ```suggestion
A performance issue in the Tooling API causing delays at the end of task execution in large projects has been identified and fixed.
```

> The issue occurred when the task information for the executed tasks was being sent to the IDE.

Reviewer: ```suggestion
This problem occurred while transmitting task information for executed tasks to the IDE.
```

Additionally, for JUnit5 and Spock, we updated the test descriptor for dynamic and parameterized tests to include information about the class name and method name containing the test.
These enhancements enable IDEs to offer improved navigation and reporting capabilities for dynamic and parameterized tests.

#### Fix IDE performance issues with large projects

A performance issue in the Tooling API causing delays at the end of task execution in large projects has been identified and fixed by a community member.
This problem occurred while transmitting task information for executed tasks to the IDE. 

> After the execution of around 15,000 tasks or more, the IDE would experience a delay of several seconds.

Reviewer: ```suggestion

After executing approximately 15,000 tasks, the IDE would encounter a delay of several seconds. 
The root cause was that much more information than needed was serialized via the Tooling API.
```

> Additionally to the fix, we added a test that ensures there will be no regression in the future. We could demonstrate a performance improvement of around 12%.

Reviewer: ```suggestion
We added a test to the fix to ensure there will be no future regression, demonstrating a performance improvement of around 12%.
```

> The root cause was that a lot more information then needed was serialized via the Tooling API.

Reviewer: ```suggestion
```
